colcon-test/action.yaml

```
@@ -0,0 +1,134 @@
+      shell: bash
+
+    - name: Set up colcon-mixin
+      run: |
+        colcon mixin add default https://raw.githubusercontent.com/colcon/colcon-mixin-repository/master/index.yaml || true
+        colcon mixin update default
+      shell: bash
+
+    - name: Install packages for coverage commands
+      run: |
+        sudo apt-get -yqq update
+        sudo apt-get -yqq install lcov python3-pip
+        pip3 install colcon-lcov-result colcon-coveragepy-result
+      shell: bash
+
+    - name: Set git config
+      if: ${{ inputs.build-depends-repos != '' }}
+      uses: autowarefoundation/autoware-github-actions/set-git-config@tier4/proposal
+      with:
+        token: ${{ inputs.token }}
+
+    - name: Clone dependency packages
+      if: ${{ inputs.build-depends-repos != '' }}
+      run: |
+        mkdir dependency_ws
+        vcs import dependency_ws < ${{ inputs.build-depends-repos }}
+      shell: bash
+
+    - name: Run rosdep install
+      run: |
+        sudo apt-get -yqq update
+        rosdep update
+        DEBIAN_FRONTEND=noninteractive rosdep install -yqq --from-paths . --ignore-src --rosdistro ${{ inputs.rosdistro }}
+      shell: bash
+
+    - name: Restore build files from cache
+      id: restore-build-files
+      uses: actions/cache@v2
+      with:
+        path: |
+          ./build
+          ./install
+        key: build-${{ github.sha }}
+
+    - name: Test
+      run: |
+        set -e
+        . /opt/ros/${{ inputs.rosdistro }}/setup.sh
+
+        colcon lcov-result --zero-counters \
+          --verbose \
+          --packages-above ${{ inputs.target-packages }}
+
+        colcon lcov-result --initial \
+          --verbose \
+          --packages-above ${{ inputs.target-packages }}
+
+        colcon test --event-handlers console_cohesion+ \
+          --mixin coverage-pytest \
+          --packages-above ${{ inputs.target-packages }} \
+          --ctest-args -L "${{ inputs.label }}" \
+          --return-code-on-test-failure
+
+        colcon lcov-result \
+          --verbose \
+          --packages-above ${{ inputs.target-packages }} || true
+
+        colcon coveragepy-result \
+          --verbose \
+          --packages-above ${{ inputs.target-packages }} || true
+      shell: bash
+
+    - name: Check lcov coverage file exists
+      id: lcov-coverage-file-existence
+      uses: autowarefoundation/autoware-github-actions/check-file-existence@tier4/proposal
+      with:
+        files: lcov/total_coverage.info
+        condition: and
+
+    - name: Check coveragepy coverage file exists
+      id: coveragepy-coverage-file-existence
+      uses: autowarefoundation/autoware-github-actions/check-file-existence@tier4/proposal
+      with:
+        files: coveragepy/.coverage
+        condition: and
+
+    - name: Generate output
+      id: test
+      run: |
+        coverage_files=()
+        if [[ ${{ steps.lcov-coverage-file-existence.outputs.exists == 'true' }} ]]; then
```

Maybe this? :thinking: 

```suggestion
        if [[ "${{ steps.lcov-coverage-file-existence.outputs.exists }}" == "true" ]]; then
```

I think this will be evaluated like this.

```bash
if [[ false ]]; then
```

But it will be `true`

```sh-session
$ [ false ] && echo 1
1

$ [ "false" ] && echo 1
1

$ [[ false ]] && echo 1
1

$ [[ "false" ]] && echo 1
1
```

---

2fd24fc